Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -31,9 +31,36 @@ public override object Deserialize(IIonReader reader)
             IonType ionType;
             while ((ionType = reader.MoveNext()) != IonType.None)
             {
-                var property = FindProperty(reader.CurrentFieldName);
+                MethodInfo method;
+                PropertyInfo property;
                 FieldInfo field;
-                if (property != null)
+
+                // Check if current Ion field has a IonPropertySetter annotated setter method.
+                if ((method = FindSetter(reader.CurrentFieldName)) != null)
+                {
+                    // A setter should be a void method.
+                    if (method.ReturnParameter?.ParameterType != typeof(void))
+                    {
+                        continue;
+                    }
+
+                    // A setter should have exactly one argument.
+                    var parameters = method.GetParameters();
+                    if (parameters.Length != 1)
+                    {
+                        continue;
+                    }
+
+                    var deserialized = ionSerializer.Deserialize(reader, parameters[0].ParameterType, ionType);
+                    if (options.IgnoreDefaults && deserialized == default)
+                    {
+                        continue;
```

Comment:
`IgnoreDefaults` does not apply to methods.

Reply:
Addressed.